Clojure/tests/syntax_test_clojure.clj

```diff
@@ -0,0 +1,2007 @@
+; SYNTAX TEST "Packages/Clojure/Clojure.sublime-syntax"
+
+; # Comments and whitespace
+
+  ;blah
+; ^ comment.line.clojure punctuation.definition.comment
+;  ^^^^ comment.line.clojure
+
+  ;;; blah
+; ^^^ comment.line.clojure punctuation.definition.comment
+;    ^^^^^ comment.line.clojure
+
+  blah;blah;blah
+; ^^^^- comment
+;     ^ comment.line.clojure
+
+  #!blah
+; ^^ comment.line.clojure punctuation.definition.comment
+;   ^^^^^ comment.line.clojure
+  #! blah
+; ^^ comment.line.clojure punctuation.definition.comment
+;   ^^^^^^ comment.line.clojure
+  #!#!#! blah
+; ^^ comment.line.clojure punctuation.definition.comment
+;   ^^^^^^^^^^ comment.line.clojure
+
+  blah,blah, blah
+;     ^ comment.punctuation.comma.clojure
+;      ^- comment
+;          ^ comment.punctuation.comma.clojure
+;           ^- comment
+
+; ## Include end-of-line
+
+; ; blah
+;^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.clojure
+
+
+
+; # Constants
+
+  true false nil
+; ^^^^ constant.language.clojure
+;     ^- constant
+;      ^^^^^ constant.language.clojure
+;           ^- constant
+;            ^^^ constant.language.clojure
+
+; ## Breaks
+
+  true,false,nil
+; ^^^^ constant.language.clojure
+;     ^ comment.punctuation.comma.clojure
+  true;false;nil
+; ^^^^ constant.language.clojure
+;     ^ comment.line.clojure punctuation.definition.comment
+
+; ## Unaffected
+
+  'nil (true) (nil)
+; ^ constant.other.symbol.reader-macro.clojure
```

Comment:
Can you explain a little bit about what sort of semantics these `constant.other.symbol.reader-macro` symbols are? It seems like a funny scope name to me, however I am not familiar with Clojure, and only slightly familiar with lisp-style languages.

Reply:
Come to think of, this scope name is probably wrong. Reader macros are basically unary prefix operators that modify the next piece of AST as it's being read. I didn't want to mark them as regular operators since they work at parse time, not runtime, and it's nice if the color scheme can make a distinction, hence the current name. In many syntaxes, `constant.other.symbol` receives special highlighting. But we should probably change this.

Reply:
Not sure if `keyword.macro.clojure` or `keyword.operator.macro.clojure` makes sense? Looks like these fall under `keyword` variety, and I want a scope that's different from `keyword.operator` but not too Clojure-specific. What would you suggest?